client/src/main/java/org/apache/uniffle/client/impl/ShuffleReadClientImpl.java

```diff
@@ -269,6 +269,10 @@ private int read() {
     // because PlatformDependent.freeDirectBuffer can only release the ByteBuffer with cleaner.
     if (sdr != null) {
       sdr.release();
+      // We set sdr to null here to prevent IllegalReferenceCountException that could occur
+      // if sdr.release() is called multiple times in the close() method,
+      // when an exception is thrown by clientReadHandler.readShuffleData().
```

Comment:
I can't get why this will trigger the issue. sdr will be reassign in the Line 277.

Reply:
> when an exception is thrown by clientReadHandler.readShuffleData().

An exception is thrown from line 277.

Reply:
OK. I know.